### DISCUSSION TITLE
Array and Plugin: Adding new items into an  array

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#386] [Q&A]
I am currently playing with arrays very successfully within the RX plugin. I am able to search within them and update existing entries.

- I can't find a way to extend an array and add new entries. This could be useful for appending an array with another array or adding some new stuff. This is not asking to live or die!
- Can we distinguish between the maximum size of an array and the maximum number of elements which are set?

**Comment by adesutherland** [comment]
Please try
GETNUMATTRS(attr) - Gets the number of attributes within a parent attribute

SETNUMATTRS(attr,num) - Sets the number of attributes of a parent attribute

The set function will (should) add elements, you can then get the element and set it's value

A

**Reply by Peter-Jacob** [reply]
runs like a charm! Thank you!

**Comment by Peter-Jacob** [comment]
Perfect, try it tomorrowTx Peter Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Adrian Sutherland ***@***.***> Datum: 21.11.24  20:43  (GMT+01:00) An: adesutherland/CREXX ***@***.***> Cc: Peter-Jacob ***@***.***>, Author ***@***.***> Betreff: Re: [adesutherland/CREXX] Array and Plugin: Adding new items into an  array (Discussion #386) 
Please try
GETNUMATTRS(attr) - Gets the number of attributes within a parent attribute
SETNUMATTRS(attr,num) - Sets the number of attributes of a parent attribute
The set function will (should) add elements, you can then get the element and set it's value
A
